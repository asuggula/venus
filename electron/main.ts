import { app, BrowserWindow } from "electron";
import * as path from "path";
import * as url from "url";

let mainWindow: Electron.BrowserWindow | null;

function createWindow() {
  mainWindow = new BrowserWindow({
    width: 800,
    height: 600,
    webPreferences: {
      nodeIntegration: true,
      webSecurity: false,
      // stops console from opening upon load
      devTools: false,
    },
  });

  if (process.env.NODE_ENV === "development") {
    mainWindow.loadURL(`http://localhost:4000`);
    mainWindow.webContents.openDevTools();
  } else {
    mainWindow.loadURL(
      url.format({
        pathname: path.join(__dirname, "index.html"),
        protocol: "file:",
        slashes: true,
      })
    );
  }

// Quit when all windows are closed, except on macOS. There, it's common
// for applications and their menu bar to stay active until the user quits
// explicitly with Cmd + Q.
app.on('window-all-closed', function () {
  if (process.platform !== 'darwin') app.quit()
})
}

app.on("ready", createWindow);

app.on('activate', function () {
  // On macOS it's common to re-create a window in the app when the
  // dock icon is clicked and there are no other windows open.
  if (BrowserWindow.getAllWindows().length === 0) createWindow()
})

app.allowRendererProcessReuse = true;
