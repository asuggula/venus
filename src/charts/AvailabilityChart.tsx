/**
 * @name AvailabilityChart
 * @desc Chart that renders historical statistics for availability on the Historical Status tab. Child component of ChartContainer
 */
import React, { useState, useEffect, useContext } from 'react';
import Line from'@ant-design/charts/es/line';
import { historicalContext } from '../contexts/historicalContext';     
const Availability: React.FC = () => {
  //set context api that broadcasts to components wrapped within it to update historical state
  const { aggregate, serviceData } = useContext(historicalContext)
  //set ContextAPI state object
  const [data, setData] = useState([]);
  let config = {
    data: serviceData,
    xField: 'year',
    yField: 'value',
    seriesField: 'category',
    xAxis: { type: 'time' },
    yAxis : { 
      label: {
        formatter: function formatter(v:any) {
          return ''.concat(v).replace(/\d{1,3}(?=(\d{3})+$)/g, function (s) {
            return ''.concat(s, ',');
          });
        },
      },
    },
  };
  console.log(config.data, 'config')
  if (config.data === undefined) return <div>loading</div>
  return <Line {...config} />;
};

export { Availability } ;  
