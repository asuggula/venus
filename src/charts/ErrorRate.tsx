/**
 * @name ErrorRate
 * @desc Chart that displays in historical status page rendering historical load statistics. Child component of ChartContainer.
 */

import React, { useState, useEffect, useContext } from "react";
import Area from "@ant-design/charts/es/area";
import { historicalContext } from "../contexts/historicalContext";
function ErrorRate(): JSX.Element {
  const { serviceData } = useContext(historicalContext);
 
  var config = {
    data: serviceData["lastHour"]["error-rate"],
    xField: "time",
    yField: "value",
    seriesField: "method",
  };
  return <Area {...config} autoFit={true} />;
}
export { ErrorRate };
