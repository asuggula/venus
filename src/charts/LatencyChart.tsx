/**
 * @name LatencyChart
 * @desc Chart that renders historical statistics for availability on the Historical Status tab. Child component of ChartContainer
 */
import React, { useContext } from 'react';
import Line from'@ant-design/charts/es/line';
import { historicalContext } from '../contexts/historicalContext';     
const Latency: React.FC = () => {
  
  const { aggregate, serviceData } = useContext(historicalContext)
  
  let config = {
    data: serviceData.response_time,
    xField: "timestamp",
    yField: 'value',
    seriesField: 'service',
    xAxis: { 
      label: {
         formatter: function formatter(v:any) {
           return v.slice(11,19); 
         }
      }
     },
    yAxis : { 
      label: {
        formatter: function formatter(v:any) {
          return ''.concat(v).replace(/\d{1,3}(?=(\d{3})+$)/g, function (s) {
            return ''.concat(s, ',');
          });
        },
        min: 500
      },
    },
    animation: {
      appear: {
        animation: 'path-in',
        duration: 1000,
      },
    },
  };
  console.log(config.data, 'config responseTime')
  return <Line {...config} />;
};

export { Latency } ;  
