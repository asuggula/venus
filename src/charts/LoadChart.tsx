/**
 * @name LoadChart
 * @desc Chart that displays in historical status page rendering historical load statistics. Child component of ChartContainer.
 */

import React, { useState, useEffect, useContext } from "react";
import Area from "@ant-design/charts/es/area";
import { historicalContext } from "../contexts/historicalContext";
function LoadChart(): JSX.Element {
const { timeRange } = useContext(historicalContext);
  console.log(timeRange.load)
  var config = {
    data: timeRange.load,
    xField: "time",
    yField: "value",
    seriesField: "method",
  };
  return <Area {...config} autoFit={true} />;
}
export { LoadChart };
