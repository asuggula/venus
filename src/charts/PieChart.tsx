import React, { useState , useEffect,useContext } from 'react' ;    
import Pie  from '@ant-design/charts/es/pie' ;    
import { liveData, historicalContext} from '../contexts/historicalContext' ;  
// import liveData from "../contexts/historicalContext";

function PieChart(): any {      
  // const { registerTheme } = G2 ;   
  // registerTheme ( 'custom-theme' , { 
  //   colors10 : [ '#FACDAA' , '#F4A49E' , '#EE7B91' , '#E85285' , '#BE408C' , '#BE408C' ] ,      
  //   colors20 : [ '#FACDAA' , '#F4A49E' , '#EE7B91' , '#E85285' , '#BE408C' , '#BE408C' , '#942D93' ] ,       
  // } ) ;
  console.log('hitting the pie chart', liveData.historical) 
  const data: any = liveData.historical

  var config = {
    appendPadding: 10,
    data: data,
    angleField: 'latency',
    colorField: 'uptime',
    radius: 0.8,
    label: {
      uptime: 'outer',
      content: '{name} {percentage}',
    },
    interactions: [{ uptime: 'pie-legend-active' }, { uptime: 'element-active' }],
  };
  return <Pie {...config} />;
};

export default PieChart ;  