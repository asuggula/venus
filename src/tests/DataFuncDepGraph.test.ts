import { ipcRenderer } from "electron"
import { changeChildArr, changeData, test, treeData } from "../charts/DataFuncDepGraph"

describe('hitting test files', () => {
  it('hits the jest testing', () => {
    expect(true).toBeTruthy();
  })
}) 

it('the dummy data matches the snapshot', () => {
  expect(test).toMatchSnapshot();
})
console.log(test)
