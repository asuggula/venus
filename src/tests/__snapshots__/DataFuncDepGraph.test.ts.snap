// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the dummy data is correct 1`] = `
Object {
  "services": Array [
    Object {
      "availability": 100,
      "byMethod": Object {
        "GET": Object {
          "availability": 100,
          "error": 0,
          "load": "0.33 hpm",
          "response_time": 1200,
          "status": "good",
        },
      },
      "error": 0,
      "load": "1.00 hpm",
      "response_time": 1266,
      "service": "gogle api",
      "status": "good",
    },
    Object {
      "availability": 100,
      "byMethod": Object {
        "GET": Object {
          "availability": 100,
          "error": 0,
          "load": "0.33 hpm",
          "response_time": 1200,
          "status": "bad",
        },
        "POST": Object {
          "availability": 100,
          "error": 0,
          "load": "0.33 hpm",
          "response_time": 1200,
          "status": "bad",
        },
      },
      "error": 0,
      "load": "1.00 hpm",
      "response_time": 1266,
      "service": "wizard api",
      "status": "fair",
    },
  ],
}
`;
